### DISCUSSION TITLE
Vitamins

### DISCUSSION THREAD

**Discussion by charos529** [nlmixrdevelopment/nlmixr#642] [Q&A]
Hello . Has smn developed PK models for vitamins with long half-life . I have population average data .I tried to use NCA to estimate parameters and then used nls function . But I am getting negative clearence and very big Ka .

**Comment by mattfidler** [comment]
I believe that people have done this before.  `nlmixr2` doesn't currently support `nls`, but you could try with `focei` if you want to use `nlmixr`.

**Comment by charos529** [comment]
I tried to fit in one- compartmental model with focei but it gives me an error .I am using population mean concentrations .

**Reply by mattfidler** [reply]
Without more information I don't believe I can help with this problem.

**Reply by charos529** [reply]
Please let me know if you need more information.Thanks a lot

**Comment by charos529** [comment]
I have the following time&conc data for Vitamin B1 marker (TDP) . 

Time (h) | DV
-- | --
0.035637 | 51.8724
0.526572 | 52.6353
1.016606 | 58.44136
1.596569 | 60.65288
2.042347 | 64.29373
2.532639 | 68.65888
3.023317 | 70.86269
3.559537 | 68.0272
4.049958 | 71.67191
6.058598 | 73.28649
8.022725 | 74.17676
12.04142 | 69.48097
23.91547 | 74.83031


I cannot estimate initial parameters bcz both NCA and NLS are not working . The plot shows that they did not take enough samples in the elimination phase I think.

**Reply by mattfidler** [reply]
I agree, sometimes that is all the feedback you can give.  If you can't get a model, you probably shouldn't force it:

``` r
library(tidyverse)
d <- tribble(~time, ~dv,
                    0.035637,   51.8724,
                    0.526572,   52.6353,
                    1.016606,   58.44136,
                    1.596569,   60.65288,
                    2.042347,   64.29373,
                    2.532639,   68.65888,
                    3.023317,   70.86269,
                    3.559537,   68.0272,
                    4.049958,   71.67191,
                    6.058598,   73.28649,
                    8.022725,   74.17676,
                    12.04142,   69.48097,
                    23.91547,   74.83031)

ggplot(d, aes(time, dv)) + geom_point()
```

![](https://i.imgur.com/mOhwoel.png)

<sup>Created on 2023-06-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by mattfidler** [reply]
There is no way I can say with any sort of certainty that there is any elimination of the vitamin happening.  You cannot fix broken data with a model no matter how hard you try.

**Reply by mattfidler** [reply]
You could fit the data with a sigmoid function but that would say that the total vitamin level would plateau, and that is making an assumption that you and I don't believe.  Probably should just circle back and say I cannot do anything with this, (sorry)

**Reply by charos529** [reply]
So should I leave it off then ?

**Comment by charos529** [comment]
I am working on PK modeling for many vitamins and  using population average data . First , I find initial estimates by NCA , then use  NLS (one compartment oral adm model) .Although I get initial estimates and run NLS successfully , the model I developed does not provide a good fit . Please can you advise me where I am being wrong .
I am just a beginner , so many apologies if I do make silly mistakes .



ID | Time | DV
-- | -- | --
1 | 0 | 75.4262
1 | 0.868538 | 184.3509
1 | 1.802486 | 211.8462
1 | 2.834105 | 301.0627
1 | 3.842697 | 234.2776
1 | 5.869877 | 168.4224
1 | 11.97609 | 138.0015
1 | 24.01126 | 76.28968



CODE :


vitaminA_paper16 <- read.csv("paper16.csv")
str(vitaminA_paper16)
library(ggplot2)
VitaminA_pl.pp16 <- ggplot(data = vitaminA_paper16, aes(x = Time, y = DV )) +
  geom_line(color = "black", linetype = "solid",alpha = 0.5) +
  theme_bw()+
  labs(x = "Time(days)", y = "Concentration in mcmol/L") +
  ggtitle("Concentration vs. Time Curve for Thiamine diphosphate")+
  geom_point(color = "blue", size = 3, alpha = 0.4)+
  theme(plot.background = element_rect(fill = "white"))
VitaminA_pl.pp16
# Noncompartmental analysis to find inital estimates 
library(ncappc)
# 1. make a placeholder for the results
VitA_pp16_out.nca <- vitA_pp16.result
#NCA analysis 
vitA_pp16.result <- est.nca(time = vitaminA_paper16$Time , 
                            conc = vitaminA_paper16$DV, 
                            doseAmt = 38.5, extrapolate = TRUE, method = "linearup-logdown")
VitA_pp16_out.nca <- rbind(VitA_pp16_out.nca,vitA_pp16.result)
#results 
VitA_pp16_out.nca
#I assumed the Bioavailability to be 1 initially 
# Define the function for the one-compartment oral administration model and perform parameter estimation using least squares  
VitA_one_cmt_model_pp16 <- nls(DV ~ (ka* 38.5 * 0.2 / (vd * ka- cl)) *
                                 (exp(-cl / vd * Time) - exp(-ka * Time)), # 1 compartment oral equation
                               start = list(ka = 1, cl = 0.0074, vd = 0.16), # inital estimates of the parameters from NCA 
                               data = vitaminA_paper16) # data to use summary(VitA_one_cmt_model_pp14)
VitA_param_estimates_pp16 <- coef(VitA_one_cmt_model_pp16)
VitA_param_estimates_pp16
#creates a sequence of numbers from 0 to 25hours with a step size of 0.1h and assign it to the variable pred.times.
pred.times_VitA_pp16 <- c(seq(0, 24, 0.1))
#extracts the estimated coefficients from the summary of the one_cmt_model_pt1 model. 
#Specifically, it assigns the estimated values of ka, cl, and vd to the variables ka_p, cl_p, and vd_p, respectively.
ka_p <- summary(VitA_one_cmt_model_pp16)$coeff[1, 1]
cl_p <- summary(VitA_one_cmt_model_pp16)$coeff[2, 1]
vd_p <- summary(VitA_one_cmt_model_pp16)$coeff[2, 1]
pred.concs_VitA_pp16 <- (ka_p * 38.5 * 0.2 / (vd_p * ka_p - cl_p)) * (exp(-cl_p / vd_p * pred.times_VitA_pp16) - exp(-ka_p * pred.times_VitA_pp16))#plot the observed time (vitaminA_combm$Time) against the observed drug concentrations (vitaminA_combm$DV) using the plot function.
plot(vitaminA_paper16$Time, vitaminA_paper16$DV)

#overlay a line representing the predicted concentrations based on the estimated model coefficients.
lines(pred.times_VitA_pp16, pred.concs_VitA_pp16, col = "blue", lwd = 2)


Model parameters from NLS 
         ka          cl          vd 
1.201202981 0.001776767 0.026376895

**Reply by mattfidler** [reply]
I'm sorry @charos529 

These functions have nothing to do with `nlmixr` or `nlmixr2` and I am not familiar enough with them to provide any advice other than what I have said: I don't believe the data have enough information to adequately describe the pharmacokinetics of the Vitamin.
